Automattic/wp-calypso
# Add-Ons: Use Purchases from data-stores for finding purchased add-ons

Fixes https://github.com/Automattic/wp-calypso/issues/87500

## Proposed Changes

Part of preparing the add-ons metadata to be derived off the `data-stores/add-ons` store. 

- We refactor the use of Calypso state for finding the purchased add-on products by utilising queries/hooks from `packages/data-stores/purchases` store.
- (Not sure about this now) We introduce a `useSitePurchasesByQuery` hook that works generically over a query that is a `Partial< Purchase >` object. It returns the same indexed structure as the underlying `useSitePurchases` query data, only filtered to those matching the query.
    - **Good:** It sounds like a great "generalised selector" to introduce in all of these new stores we are working on (Plans, Purchases, Products, Site Features, etc.)
    - **Bad:** `query` being an object, it will need to be a static reference to allow memoizing of the results returned by the selector. It makes usage complicated if we want referential integrity.

^ will most likely be rewriting this from the looks of it. Not sure why I haven't thought about the drawbacks before putting it together 🤷‍♂️ 

**EDIT:** Retracted the above, for the reasons outlined. Converted it into a `getSitePurchasesByProductSlug` selector. It may not even be necessary, but also no harm as either or (e.g. doing it inside the component) the code is needed.

## Testing Instructions



* Confirm a `purchased` label is shown in `/add-ons` when an add-on has been purchased already

## Pre-merge Checklist
